rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/NodeSourceWindow.java

```diff
@@ -0,0 +1,473 @@
+            PluginDescriptor.Field pluginField);
+
+    protected String getPluginShortName(PluginDescriptor plugin) {
+
+        return plugin.getPluginName().substring(plugin.getPluginName().lastIndexOf('.') + 1);
+    }
+
+    protected ArrayList<FormItem> prepareFormItems() {
+
+        this.infrastructureSelectItem = new SelectItem("infra", "Infrastructure");
+        this.infrastructureSelectItem.setRequired(true);
+        this.policySelectItem = new SelectItem("policy", "Policy");
+        this.policySelectItem.setRequired(true);
+
+        this.infrastructureSelectItem.setWidth(300);
+        this.policySelectItem.setWidth(300);
+
+        HiddenItem name = new HiddenItem("nsName");
+        HiddenItem nodesRecoverable = new HiddenItem("nodesRecoverable");
+        HiddenItem deploy = new HiddenItem("deploy");
+        HiddenItem nodeSourceEdited = new HiddenItem("nodeSourceEdited");
+        HiddenItem callback = new HiddenItem("nsCallback");
+        HiddenItem session = new HiddenItem("sessionId");
+
+        ArrayList<FormItem> formItems = new ArrayList<>();
+
+        formItems.add(name);
+        formItems.add(nodesRecoverable);
+        formItems.add(deploy);
+        formItems.add(nodeSourceEdited);
+        formItems.add(callback);
+        formItems.add(session);
+
+        return formItems;
+    }
+
+    protected ArrayList<FormItem> getPrefilledFormItems(PluginDescriptor plugin) {
+
+        List<PluginDescriptor.Field> pluginFields = plugin.getConfigurableFields();
+        ArrayList<FormItem> formItems = new ArrayList<>(pluginFields.size());
+
+        List<FormItem> formItemsForField = new LinkedList<>();
+        for (PluginDescriptor.Field pluginField : pluginFields) {
+
+            if (pluginField.isPassword()) {
+
+                formItemsForField.add(new PasswordItem(plugin.getPluginName() + pluginField.getName(),
+                                                       pluginField.getName()));
+
+            } else if (pluginField.isFile() || pluginField.isCredential()) {
+
+                formItemsForField.addAll(handleNonTextualPluginField(plugin, pluginField));
+
+            } else {
+
+                formItemsForField.add(new TextItem(plugin.getPluginName() + pluginField.getName(),
+                                                   pluginField.getName()));
+            }
+
+            formItemsForField.forEach(formItem -> {
+                formItem.setValue(pluginField.getValue());
+                formItem.setWidth(250);
+                if (!formItem.getName().equals(NodeSourceEditWindow.KEEP_OR_CHANGE_FORM_ITEM_NAME)) {
+                    formItem.setHint("<nobr>" + pluginField.getDescription() + "</nobr>");
+                }
+            });
+
+            formItems.addAll(formItemsForField);
+            formItemsForField.clear();
+        }
+
+        return formItems;
+    }
+
+    // TODO Method to use when the server is ready to deal with
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)